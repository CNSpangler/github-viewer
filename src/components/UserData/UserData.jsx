import React from 'react';
import PropTypes from 'prop-types';

const UserData = ({ username, followers, following, profile }) => (
  <>
    <section>
      <h2>{username}</h2>
      <h3>Github Profile: {profile}</h3>
      <h4>Followers: {followers}</h4>
      <h4>Following: {following}</h4>
    </section>
  </>
);

UserData.propTypes = {
  username: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  profile: PropTypes.string.isRequired
};

export default UserData;
