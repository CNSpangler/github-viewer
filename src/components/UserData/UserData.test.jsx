import React from 'react';
import { shallow } from 'enzyme';
import UserData from './UserData.jsx';

describe('UserData component', () => {
  it('renders UserData', () => {
    const wrapper = shallow(
      <UserData 
        username="cnspangler"
        profile="https://api.github.com/users/cnspangler"
        followers="10"
        following="25"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
