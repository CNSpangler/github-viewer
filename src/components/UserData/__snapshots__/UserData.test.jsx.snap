// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserData component renders UserData 1`] = `
<Fragment>
  <section>
    <h2>
      cnspangler
    </h2>
    <h3>
      Github Profile: 
      https://api.github.com/users/cnspangler
    </h3>
    <h4>
      Followers: 
      10
    </h4>
    <h4>
      Following: 
      25
    </h4>
  </section>
</Fragment>
`;
